webapp/src/js/services/enketo.js

```diff
@@ -274,19 +268,41 @@ angular.module('inboxServices').service('Enketo',
         overrideNavigationButtons(currentForm, wrapper);
         addPopStateHandler(currentForm, wrapper);
         forceRecalculate(currentForm);
-
+        setupNavButtons(currentForm, wrapper, 0);
         return currentForm;
       });
     };
 
+    var setupNavButtons = function(form, $wrapper, currentIndex) {
+      if(form.pages) {
+        var lastIndex = form.pages.$activePages.length - 1;
+        var footer = $wrapper.find('.form-footer');
+        if( currentIndex >= lastIndex ) {
+          footer.addClass('end').find('.next-page').addClass('disabled');
+          if( currentIndex === 0 ) {
+            footer.find('.previous-page').addClass('disabled');
+          }
+        } else if( currentIndex === 0 ) {
+          footer.find('.previous-page').addClass('disabled');
```

Comment:
This is duplicated with the section above. Pull the `if (currentIndex === 0) {...}` block out and run that first so you don't need to duplicate it.

Reply:
Done.